### PR TITLE
Blokh/feat/workflow definition mt

### LINKED ISSUE #1247: workflow definition mt

### PR DESCRIPTION
**Pull request by Blokh**
### Description
added workflow definition clone logic and scoping of workflow definition

### CODE REVIEW

File: services/workflows-service/src/workflow/workflow-definition.repository.ts [hunk 1]
```diff
@@ -45,6 +45,19 @@ export class WorkflowDefinitionRepository {
...
 
+  async findTemplateByIdUnscoped<
+    T extends Omit<Prisma.WorkflowDefinitionFindFirstOrThrowArgs, 'where'>,
+  >(
+    id: string,
+    args: Prisma.SelectSubset<T, Omit<Prisma.WorkflowDefinitionFindFirstOrThrowArgs, 'where'>>,
+    projectId?: string,
+  ): Promise<WorkflowDefinition> {
+    return await this.prisma.workflowDefinition.findFirstOrThrow({
+      where: { id, isTemplate: true, projectId: projectId ?? null },
```
Comment (MatanYadaev) [inline services/workflows-service/src/workflow/workflow-definition.repository.ts]: Remove `projectId` as it's irrelevant here.
```suggestion
  ): Promise<WorkflowDefinition> {
    return await this.prisma.workflowDefinition.findFirstOrThrow({
      where: { id, isTemplate: true },
```

File: services/workflows-service/src/workflow/workflow.service.ts [hunk 2]
```diff
@@ -149,9 +150,40 @@ export class WorkflowService {
...
+    };
+
+    const workflowDefinition = await this.workflowDefinitionRepository.findTemplateByIdUnscoped(
+      data.id,
+      { select },
+    );
+
+    return await this.workflowDefinitionRepository.create(
+      // @ts-expect-error - types of workflow definition does not propagate to the prisma creation type
+      { data: { ...workflowDefinition, name: data.name }, select },
```
Comment (MatanYadaev) [inline services/workflows-service/src/workflow/workflow.service.ts]: `workflowDefinition` has an `id` property, it won't use it?

**Review by MatanYadaev — APPROVED** [review]

File: services/workflows-service/src/workflow/workflow.service.ts (hunk 2 above)
Comment (Blokh) [reply services/workflows-service/src/workflow/workflow.service.ts]: nop, it is not being selected in the select args.

File: services/workflows-service/src/workflow/workflow-definition.repository.ts (hunk 1 above)
Comment (Blokh) [reply services/workflows-service/src/workflow/workflow-definition.repository.ts]: I somewhat agree.
but there's some kind of logic of creating "templates" (publics) to specific customers.
e.g. nuvei will have multiple of the same template

**Pull request merged**